File: domain/src/main/kotlin/io/landzcape/dto/ComponentDto.kt
package io.landzcape.dto

data class ComponentDto(val name: String,
                        val group: String,
                        val version: String,
                        val label: String,
                        val layer: String? = null,
                        val domain: String? = null,
                        val context: String? = null,
                        val type: String = ComponentTypeDto.BUSINESS.name,
                        val interfaces: List<DependencyDto> = listOf(),
                        val dependencies: List<DependencyDto> = listOf()
) {
}

File: explorer/webapp/src/app/api/model/component.dto.ts
import {DependencyDto} from './dependency.dto';

export class ComponentDto {
  name: string;
  group: string;
  version: string;
  label: string;
  layer: string;
  domain: string;
  context: string;
  type: string;
  interfaces: DependencyDto[];
  dependencies: DependencyDto[];
}
